File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {

        int n = nums.size();
        unordered_map<int,int> freq;

        for(int i:nums) freq[i]++;

        for(auto it=freq.begin();it!=freq.end();it++){
            if(it->second > n/2) return it->first;
        }
        return 0;
    }
};

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        
        counts = {}
        size = len(nums) // 2

        for i in nums:
            if i in counts:
                counts[i] += 1
            else:
                counts[i] = 1
            if counts[i] > size:
                return i